nikola/plugin_categories.py

```diff
@@ -336,12 +336,12 @@ def split_metadata(self, data):
         """Split data from metadata in the raw post content.
 
         This splits in the first empty line that is NOT at the beginning
-        of the document.
+        of the document, or after YAML/TOML metadata without an empty line.
         """
         if data.startswith('---'):  # YAML metadata
-            split_result = re.split('(\n---\n\n|\r\n---\r\n\r\n)', data.lstrip(), maxsplit=1)
```

Comment:
It’s not necessary if we have a `---` / `+++` separator already. This also mimics Jekyll behavior.